core/services/keeper/upkeep_executer.go

```diff
@@ -278,7 +278,7 @@ func (ex *UpkeepExecuter) estimateGasPrice(upkeep UpkeepRegistration) (gasPrice
 	}
 
 	if ex.config.EvmEIP1559DynamicFees() {
-		fee, _, err = ex.gasEstimator.GetDynamicFee(upkeep.ExecuteGas)
+		fee, _, err = ex.gasEstimator.GetDynamicFee(upkeep.ExecuteGas, ex.config.EvmMaxGasPriceWei())
```

Comment:
Keepers doesn't have key-specific gas lanes so this just passes in the global limit. I think letting the caller decide which limit to use instead of passing a map allows more flexibility. One alternative here is to pass nil here but went with this approach.